vite.config.js

```diff
@@ -1,11 +1,15 @@
 import { defineConfig } from 'vite';
 import reactRefresh from '@vitejs/plugin-react-refresh';
 import * as esbuild from 'esbuild';
-import path from 'path';
-import pkg from './package.json';
+import path, { dirname } from 'path';
+import { fileURLToPath } from 'url';
+import pkg from './package.json' assert { type: 'json' }; // must do the assert to solve https://nodejs.org/api/errors.html#err_import_assertion_type_missing
+
+const _dirname = fileURLToPath(dirname(import.meta.url));
```

Comment:
There is no `__dirname` when `type: module` ([see here](https://nodejs.org/api/esm.html#no-__filename-or-__dirname)), so I replace this. However, node (?) still replaces `const __dirname` with the actual dirname so I changed the variable name to `_dirname`.

```
✘ [ERROR] Expected identifier but found "\"/Users/etowah/projects/gosling-lang/gosling.js\""

    vite.config.js:8:6:
      8 │ const "/Users/etowah/projects/gosling-lang/gosling.js" = fileURLToPath(dirname("file:///Users/etowah/projects/gosling-lang/gosling.js/vite.config.js"));
```